### ISSUE TITLE
[BUG] WorkRefer can be null cause throw NPE

### ISSUE THREAD

**Issue by AngersZhuuuu** [alibaba/RemoteShuffleService#80]
### What is the bug?

Worker in blacklist and WorkerRef is null
```
22/03/29 15:01:46 WARN LifecycleManager: [reserve buffer] failed due to blacklist:
Host: xxx.xxx.xxx.xx
RpcPort: 45149
PushPort: 46795
FetchPort: 38947
TotalSlots: -1
SlotsUsed: 0
SlotsAvailable: -1
LastHeartBeat: 0
WorkerRef: null

```

Cause task throw NPE when reserveSlot 
```

22/03/29 15:01:46 ERROR Inbox: Ignoring error
java.lang.NullPointerException
	at com.aliyun.emr.rss.client.write.LifecycleManager.$anonfun$reserveSlotsWithRetry$2(LifecycleManager.scala:803)
	at com.aliyun.emr.rss.client.write.LifecycleManager.$anonfun$reserveSlotsWithRetry$2$adapted(LifecycleManager.scala:787)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at com.aliyun.emr.rss.client.write.LifecycleManager.reserveSlotsWithRetry(LifecycleManager.scala:787)
	at com.aliyun.emr.rss.client.write.LifecycleManager.handleRegisterShuffle(LifecycleManager.scala:287)
	at com.aliyun.emr.rss.client.write.LifecycleManager$$anonfun$receiveAndReply$1.applyOrElse(LifecycleManager.scala:180)
	at com.aliyun.emr.rss.common.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:110)
	at com.aliyun.emr.rss.common.rpc.netty.Inbox.safelyCall(Inbox.scala:214)
	at com.aliyun.emr.rss.common.rpc.netty.Inbox.process(Inbox.scala:107)
	at com.aliyun.emr.rss.common.rpc.netty.Dispatcher$MessageLoop.run(Dispatcher.scala:222)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


```

### How to reproduce the bug?
Close and restart a worker when  running application

### Could you share logs or screenshots?
If applicable, add logs/screenshots to help explain your problem.

/cc @who-need-to-know

/assign @who-can-solve-this-bug

**Comment by AngersZhuuuu** [comment]
Raise a pr soon

**Comment by AngersZhuuuu** [comment]
Resolved in #81
